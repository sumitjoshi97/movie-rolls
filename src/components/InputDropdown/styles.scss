.input-dropdown {
  // background: rgba(255, 255, 255, 0.1);
  // color: #fff;
  padding-left: 1rem;
  border: none;
  border-radius: 2px;
  position: relative;
  cursor: pointer;

  &:focus,
  &:active {
    border: none;
    outline: none;
  }

  &__header {
    display: flex;
    align-items: center;
    width: 9.6rem;
    color: #afafaf;
    text-transform: capitalize;

    span {
      padding-right: 4px;
    }
  }

  &__list {
    position: absolute;
    top: 2.8rem;
    left: 0;
    background: #2c2c2c;
    z-index: 400;
    overflow-y: auto;
    max-height: 29rem;
    list-style: none;
    &__option {
      padding: 0.6rem 1.6rem 0.6rem 1.2rem;
      text-transform: capitalize;
      color: #dddddd;
    }
  }
}
