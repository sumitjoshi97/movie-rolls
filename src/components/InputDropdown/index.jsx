import React, { PureComponent } from 'react'
import IosArrowDown from 'react-ionicons/lib/IosArrowDown'

import './styles.scss'

export default class InputDropdown extends PureComponent {
  state = {
    isActive: false
  }

  toggleList = () => {
    this.setState(prevState => ({
      isActive: !prevState.isActive
    }))
  }

  handleOption = (name, option) => {
    this.props.handleDropdown(name, option)
    this.toggleList()
  }

  render() {
    return (
      <div className='input-dropdown'>
        <div className='input-dropdown__header' onClick={this.toggleList}>
          <span>{this.props.value.label}</span>
          <IosArrowDown fontSize='18px' color='#ff0000' />
        </div>

        {this.state.isActive && (
          <ul className='input-dropdown__list'>
            {this.props.options.map(option => (
              <li
                className='input-dropdown__list__option'
                key={option.value}
                onClick={() => this.handleOption(this.props.name, option)}
              >
                {option.label}
              </li>
            ))}
          </ul>
        )}
      </div>
    )
  }
}
