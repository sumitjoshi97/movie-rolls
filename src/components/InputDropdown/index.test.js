import React from 'react'
import { shallow } from 'enzyme'

import InputDropdown from './index'

describe('InputDropdown component unit test', () => {
  const handleDropdown = jest.fn()
  const name = 'year'
  const options = [
    { label: '2019', value: '2019' },
    { label: '2018', value: '2018' }
  ]
  const value = {
    label: '2019',
    value: '2019'
  }

  const props = {
    name,
    options,
    value,
    handleDropdown
  }

  const inputDropdownWrapper = shallow(<InputDropdown {...props} />)

  test('will render properly', () => {
    expect(inputDropdownWrapper).toMatchSnapshot()
  })

  test('clicking on list opens dropdown', () => {
    inputDropdownWrapper.find('.input-dropdown__header').simulate('click')
    expect(inputDropdownWrapper.state().isActive).toEqual(true)
  })

  test('clicking on dropdown option calls callback fn. with options', () => {
    inputDropdownWrapper
      .find('.input-dropdown__list__option')
      .at(1)
      .simulate('click')

    expect(handleDropdown).toHaveBeenCalledWith(name, options[1])
  })
})
