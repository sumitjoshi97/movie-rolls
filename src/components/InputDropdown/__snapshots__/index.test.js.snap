// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDropdown component unit test will render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputDropdown
    handleDropdown={[MockFunction]}
    name="year"
    options={
      Array [
        Object {
          "label": "2019",
          "value": "2019",
        },
        Object {
          "label": "2018",
          "value": "2018",
        },
      ]
    }
    value={
      Object {
        "label": "2019",
        "value": "2019",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="input-dropdown__header"
          onClick={[Function]}
        >
          <span>
            2019
          </span>
          <IosArrowDown
            beat={false}
            color="#ff0000"
            fontSize="18px"
            rotate={false}
            shake={false}
            style={Object {}}
          />
        </div>,
        false,
      ],
      "className": "input-dropdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              2019
            </span>,
            <IosArrowDown
              beat={false}
              color="#ff0000"
              fontSize="18px"
              rotate={false}
              shake={false}
              style={Object {}}
            />,
          ],
          "className": "input-dropdown__header",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2019",
            },
            "ref": null,
            "rendered": "2019",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "beat": false,
              "color": "#ff0000",
              "fontSize": "18px",
              "rotate": false,
              "shake": false,
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-dropdown__header"
            onClick={[Function]}
          >
            <span>
              2019
            </span>
            <IosArrowDown
              beat={false}
              color="#ff0000"
              fontSize="18px"
              rotate={false}
              shake={false}
              style={Object {}}
            />
          </div>,
          false,
        ],
        "className": "input-dropdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                2019
              </span>,
              <IosArrowDown
                beat={false}
                color="#ff0000"
                fontSize="18px"
                rotate={false}
                shake={false}
                style={Object {}}
              />,
            ],
            "className": "input-dropdown__header",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2019",
              },
              "ref": null,
              "rendered": "2019",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "beat": false,
                "color": "#ff0000",
                "fontSize": "18px",
                "rotate": false,
                "shake": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
